lib/game_logic.rb

```diff
@@ -0,0 +1,166 @@
+# rubocop: disable Style/ClassVars, Metrics/LineLength, Style/GuardClause
+require 'colorize'
+class UserInterface
+  def self.players
+    @@players
+  end
+
+  def initialize(name1, name2)
+    @name1 = name1
+    @name2 = name2
+    @@players = []
+    @@players << @name1
+    @@players << @name2
+  end
+
+  def self.show_players
+    "you are #{@@players[0]} your mark is (X) and you choose go vs #{@@players[1]} his mark is (O)"
+  end
+
+  def self.show_game_instructions
+    "Welcome! \n This is our Tic-Tac-Toc Game \n For this game you will need THE AMAZING INSTRUCTIONS: \n The game start in a grid table of 3x3 cells with numbers.\n You will be 2 players, the player 1 play first, and the second... play second, one turn by one, choosing what number play \n You need complete a line (horizontal, vertical or diagonal) with 3 grid-cells or numbers\n".green
+  end
+
+  def self.show_who_play
+    LogicGame.who_play?
+  end
+
+  def self.choose
+    @turn = LogicGame.turn
+    "you choose #{@turn}".blue
+  end
+
+  def self.invalid_move
+    'You choose an unvalid option, please try again and select an option from the available as shown'.red
+  end
+
+  def self.show_win_game
+    if !LogicGame.player_winner.nil?
+      LogicGame.player_winner
+    else
+      "It's a draw"
+    end
+  end
+end
+
+class LogicGame
+  @@rows_cols = [[1, 2, 3], [4, 5, 6], [7, 8, 9], [1, 4, 7], [2, 5, 8], [3, 6, 9], [1, 5, 9], [3, 5, 7]]
+  @@chosen_p1 = []
+  @@chosen_p2 = []
+  @@turns_available = 9
+  @@available_moves = (1..9).to_a
+  @@available_moves_shown = @@available_moves
+  @@turn = 0
+  @@win = false
+
+  def initialize(numb)
+    @@players = numb
+    @@player_winner = []
+  end
+
+  def self.who_play?
+    if @@turns_available.odd?
+      "Ready #{@@players[0]}: \n (moves available: #{@@available_moves}".light_red
+    else
+      "Ready #{@@players[1]}: \n (moves available: #{@@available_moves}".light_green
+    end
+  end
+
+  def self.game_play(turn)
+    @@turn = turn
+    @turn = turn
+    @map_available = @@available_moves.reject { |a| a.to_i == turn.to_i }
+    if @@available_moves.length == @map_available.length || turn.zero?
+      return UserInterface.invalid_move
+    else
+      Board.board_modf(@@available_moves, @map_available, @@available_moves_shown, @@turns_available)
+      Board.gameboard
+      @@turns_available -= 1
+      @@win = comparing_players if @@chosen_p1.size >= 3
+    end
+
+    @@win
+  end
+
+  def self.comparing_players
+    @@rows_cols.each do |n_row|
+      count1 = 0
+      count2 = 0
+      n_row.each do |n|
+        if @@chosen_p1.include?(n)
+          count1 += 1
+        elsif @@chosen_p2.include?(n)
+          count2 += 1
+        else
+          count1 = 0
+          count2 = 0
+        end
+        @@win = true unless chossing_winner(count1, count2).nil?
+      end
+    end
+    @@win
+  end
```

Comment:
- [ ] `[OPTIONAL]` you can improve the winning checks, by using a "current player" argument and use _that_ to increment count1, instead of incrementing both variables, which will reduce the number of operations happening.